mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -117,11 +130,21 @@ LogicalResult Conv2dGenerator::isValidConvolution() const {
     return failure();
   }
 
+  static const llvm::StringMap<size_t> typeWidths{
+    {"f32", 4}, {"fp32", 4},
```

Comment:
How about sizeof(fp32) and avoid hard coded sizes?

Reply:
... Right, that exists

Although it does assume the sizeof(fp32) on the GPU is the same as on the host ... which is pretty reasonable.

Reply:
Fixed